lib/ood_core/job/adapters/torque.rb

```diff
@@ -201,7 +201,12 @@ def info_where_owner(owner)
         # @see Adapter#info
         def info(id)
           id = id.to_s
-          parse_job_info(*@pbs.get_job(id).flatten)
+
+          result = @pbs.get_job(id)
+
+          return parse_job_info(*result.flatten) if result.keys.length == 1
+
+          parse_job_array(id, result)
```

Comment:
So the assumptions in this code and `parse_job_array` as I understand it are:

1. `@pbs.get_job(id)` returns a hash of key value pairs, one for each job or task. So a regular job only will have one key value pair, and a job array will have multiple
2. If we are dealing with a job array, the parent task is the first hash, both returned by `result.keys.first` and `result.to_a.first`. Is this guaranteed to always be the case, or is there a better way to determine the parent task than placement in the hash (such as the id - is the id of the parent task more closely related to the id used when calling `get_job`?
3. I also noticed `parent_task.delete(:exec_host)`. It says "Master tasks don't actually run on a host" but child tasks do? The problem I see here is that for child tasks we are not preserving exec_host, only the task id and statuses. I thought the entire job array ran on the same exec_hosts or at least in the same set of nodes allocated for the job. So I imagine we would want to preserve this information when displaying job info for a single job. Could you explain what you were thinking here?

Reply:
Oh I was mistaken, from reading the code it would seem that for a job array get_jobs returns one hash per task. So the "parent task is just getting the first task's hash, and the assumption is the rest of the values are shared.

So does that mean that parent_task has a status equal to the value of the status of the first task?

Reply:
That's correct that the first hash returned is used to populate most values for the parent task. That may not be what we want. If the first job has begun executing then it may have CPU time which isn't really valid for a 'parent' job.

As for the status of a parent task: no. If there are `#tasks` then `job_array_aggregate_task_status` is called against the statuses of all child tasks.

Reply:
So instead of `parent_task.delete(:exec_host)` I imagine we would be doing something like this:

```ruby
# exec host looks like "n0163/2,7,10-11+n0205/0-11+n0156/0-11"
# assuming exec host for a non running job is nil
# assuming exec host strings from multiple nodes can be merged with a +; but if each task is running on the same node the result may be silly: "n0163/0-27+n0163/0-27+n0163/0-27+n0163/0-27" or maybe slightly informative "n0163/0+n0163/1+n0163/2+n0163/3"
parent_task[:exec_host] = results.map { |k,v| v[:exec_host] }.compact.join("+")
```

The beneficial thing about this approach is that this code works to get the exec_host of a normal job.

For example in irb:

```
irb(main):001:0> job = {"4205673.owens-batch.ten.osc.edu" => { exec_host: "o0732/0" }}
=> {"4205673.owens-batch.ten.osc.edu"=>{:exec_host=>"o0732/0"}}
irb(main):002:0> jobary =  {"4205664[1].owens-batch.ten.osc.edu" => { exec_host: "o0732/0" }, "4205664[2].owens-batch.ten.osc.edu" => { exec_host: "o0732/1"}}
=> {"4205664[1].owens-batch.ten.osc.edu"=>{:exec_host=>"o0732/0"}, "4205664[2].owens-batch.ten.osc.edu"=>{:exec_host=>"o0732/1"}}
irb(main):003:0> def aggregate_exec_host(results)
irb(main):004:1>   results.map { |k,v| v[:exec_host] }.compact.join("+")
irb(main):005:1> end
=> nil
irb(main):007:0> aggregate_exec_host jobary
=> "o0732/0+o0732/1"
irb(main):006:0> aggregate_exec_host job
=> "o0732/0"
irb(main):009:0> job["4205673.owens-batch.ten.osc.edu"][:exec_host]
=> "o0732/0"
irb(main):011:0> job["4205673.owens-batch.ten.osc.edu"][:exec_host] == aggregate_exec_host(job)
=> true
```

Reply:
If we take this approach, then two methods can be written, one to get the aggregate_exec_host and another to get the task_list and for a normal job aggregate_exec_host returns the exec_host and task_list returns [] and you won't need any special cases or control flow statements.

Reply:
Added in 148f12f.